multimap/src/test/java/org/infinispan/multimap/impl/EmbeddedMultimapListCacheTest.java

```diff
@@ -301,4 +301,48 @@ public void testPollLast() {
                   .thenAccept(r1 -> assertThat(r1).isNull())
       );
    }
+
+   public void testSubList() {
+      await(listCache.offerLast(NAMES_KEY, OIHANA));
+      await(listCache.offerLast(NAMES_KEY, ELAIA));
+      await(listCache.offerLast(NAMES_KEY, KOLDO));
+      await(listCache.offerLast(NAMES_KEY, RAMON));
+      await(listCache.offerLast(NAMES_KEY, JULIEN));
+
+      assertThat(await(listCache.subList(NAMES_KEY, 0, 0))).containsExactly(OIHANA);
+      assertThat(await(listCache.subList(NAMES_KEY, 4, 4))).containsExactly(JULIEN);
+      assertThat(await(listCache.subList(NAMES_KEY, 5, 5))).isEmpty();
+      assertThat(await(listCache.subList(NAMES_KEY, -1, -1))).containsExactly(JULIEN);
+      assertThat(await(listCache.subList(NAMES_KEY, -5, -5))).containsExactly(OIHANA);
+      assertThat(await(listCache.subList(NAMES_KEY, -6, -6))).isEmpty();
+
+      assertThat(await(listCache.subList(NAMES_KEY, 0, 4))).containsExactly(OIHANA, ELAIA, KOLDO, RAMON, JULIEN);
+      assertThat(await(listCache.subList(NAMES_KEY, 0, 5))).containsExactly(OIHANA, ELAIA, KOLDO, RAMON, JULIEN);
+      assertThat(await(listCache.subList(NAMES_KEY, 0, 3))).containsExactly(OIHANA, ELAIA, KOLDO, RAMON);
+      assertThat(await(listCache.subList(NAMES_KEY, 1, 2))).containsExactly(ELAIA, KOLDO);
+
+      assertThat(await(listCache.subList(NAMES_KEY, 1, 0))).isEmpty();
+      assertThat(await(listCache.subList(NAMES_KEY, -1, 0))).isEmpty();
+      assertThat(await(listCache.subList(NAMES_KEY, -1, -2))).isEmpty();
+
+      assertThat(await(listCache.subList(NAMES_KEY, -5, -1))).containsExactly(OIHANA, ELAIA, KOLDO, RAMON, JULIEN);
+      assertThat(await(listCache.subList(NAMES_KEY, -5, -2))).containsExactly(OIHANA, ELAIA, KOLDO, RAMON);
+      assertThat(await(listCache.subList(NAMES_KEY, -4, -2))).containsExactly(ELAIA, KOLDO, RAMON);
+
+      assertThat(await(listCache.subList(NAMES_KEY, 1, -1))).containsExactly(ELAIA, KOLDO, RAMON, JULIEN);
+      assertThat(await(listCache.subList(NAMES_KEY, 1, -1))).containsExactly(ELAIA, KOLDO, RAMON, JULIEN);
```

Comment:
A bit confusing as to why it accepts something like (1, -1) and not (1, 0). I understand this is also Redis behavior, so maybe it's a matter of documenting later.

Reply:
I figure out that there are scenarios you just want the whole list (0, -1) or skip just the first element. providing -1 means until the last element. Clients can this way avoid sending the size of the list

Reply:
1 0  since for the second element until the first one...

Reply:
https://issues.redhat.com/browse/ISPN-14896